repoman/git/depot_operations.py

```diff
@@ -50,24 +50,31 @@ def grab_changesets(self, path, url, changesets):
         logger.debug('Grabbing changesets from %s to %s' % (url, path))
         git_repo = pygit2.Repository(path)
         try:
-            origin = self._set_origin_source(git_repo, url)
-            #this may fail against github
-            result = origin.fetch()
-            logger.debug('GIT Done grabbing changesets (%s)' % (result))
-        except (pygit2.GitError, OSError) as e:
-            logger.exception('Error Grabbing changesets: %s' % e)
-            logger.info("Retrying using process")
-            try:
+            self._set_origin_source(git_repo, url)
```

Comment:
I've been trying different approcahes.
I'd add a `git reset --hard` operation as a first step, since the working copy could be dirty because of some failures in  previous tasks. In that way, it could be something like:
```
                previous_branch = git_repo.head.shorthand
                subprocess.call('git reset --hard', cwd=path, shell=True)
                subprocess.call('git checkout --detach', cwd=path, shell=True)
                subprocess.call('git fetch origin', cwd=path, shell=True)
                subprocess.call('git checkout %s' % previous_branch, cwd=path, shell=True)
```
WDYT?